app/uk/gov/hmrc/gform/sharedmodel/formtemplate/Section.scala

```diff
@@ -46,11 +47,21 @@ object Section {
     repeatsMax: Option[Expr],
     pages: NonEmptyList[Page],
     addAnotherQuestion: FormComponent,
-    instruction: Option[Instruction]
+    instruction: Option[Instruction],
+    presentationHint: Option[List[PresentationHint]]
   ) extends Section {
     override lazy val expandedFormComponents: List[FormComponent] = pages.toList.flatMap(_.expandedFormComponents)
   }
 
+  object AddToList {
+
+    implicit val presentationHintReads: Reads[List[PresentationHint]] = (json: JsValue) =>
+      parseOpt(JsDefined(json), PresentationHintParser.validate)
+        .fold(us => JsError(us.toString), fv => JsSuccess(fv.getOrElse(List.empty)))
+
+    implicit val format: Format[AddToList] = Json.format[AddToList]
+  }
+
```

Comment:
I was hoping in more subtle solution, like hiding this `Reads[List[PresentationHint]]` in `def readAddToList(json: JsValue)` and not implementing it in terms of `parseOpt`.

Something like:

```
private def readAddToList(json: JsValue) = {
  implicit def presentationHintsReads: OFormat[List[PresentationHint]] =
    OFormatWithTemplateReadFallback {
      case JsString(str) => PresentationHintParser.validate(str).fold(e => JsError(e.error), JsSuccess(_))
      case unknown       => JsError("Expected String, got " + unknown)
    }
  Json.format[Section.AddToList].reads(json)
}
```

Then we can revert `toOpt` changes etc.

Reply:
This is still a dirty solution, though. The best thing would be to introduce newtype for `List[PresentationHint]` and do the logic there.

Reply:
Implemented as per the discussion